taurus/util/impl.py

```diff
@@ -136,8 +150,8 @@ def _flatten(base_obj):
 
         return to_return
 
-    res = recursive_flatmap(obj, _flatten)
-    return [substitute_links(x) for x in res]
+    res = recursive_flatmap(obj, _flatten, chronological=True)
+    return sorted([substitute_links(x) for x in res], key=lambda x: _flattened_sort_order(x))
```

Comment:
This sorting is how we can re-link the objects when being loaded from the preamble.